wolfcrypt/src/ecc.c

```diff
@@ -6947,14 +7096,65 @@ static int wc_ecc_import_raw_private(ecc_key* key, const char* qx,
             err = wc_export_int(key->pubkey.y, &key->pubkey_raw[keySz],
                 &keySz, keySz, WC_TYPE_UNSIGNED_BIN);
     }
+#elif defined(WOLFSSL_CRYPTOCELL)
+    if (err == MP_OKAY) {
+        key_raw[0] = ECC_POINT_UNCOMP;
+        keySz = key->dp->size;
+        err = wc_export_int(key->pubkey.x, &key_raw[1], &keySz, keySz,
+            WC_TYPE_UNSIGNED_BIN);
+        if (err == MP_OKAY)
+            err = wc_export_int(key->pubkey.y, &key_raw[1+keySz],
+                &keySz, keySz, WC_TYPE_UNSIGNED_BIN);
+
+
+        pDomain = CRYS_ECPKI_GetEcDomain(cc310_mapCurve(curve_id));
+
+        /* create public key from external key buffer */
+        err = CRYS_ECPKI_BuildPublKeyFullCheck(pDomain,
+                                               key_raw,
+                                               keySz*2 + 1,
+                                               &key->ctx.pubKey,
+                                               &tempBuff);
+
+        if (err != SA_SILIB_RET_OK){
+            WOLFSSL_MSG("CRYS_ECPKI_BuildPublKeyFullCheck failed");
+            return err;
+        }
+    }
+
 #endif
 
     /* import private key */
     if (err == MP_OKAY) {
-        if (d != NULL) {
+        if (d != NULL && d[0] != '\0') {
```

Comment:
What caused the addition of the `d[0] != '\0' check?

Reply:
`d[0] != '\0' was added in few places to handle the case where an empty string is passed from the caller function instead of a NULL.